class-constructor-new/exercise-1.js

```diff
@@ -0,0 +1,73 @@
+export class Vector extends Array {
+  constructor(argIn) {
+    if (!Vector.validateConstructorParams(argIn)) {
+      throw new Error('Invalid constructor argument');
+    }
+
+    super(...argIn);
+  }
+
+  static validateConstructorParams(argIn) {
+    return Array.isArray(argIn) && argIn.length > 0 && argIn.every((i) => Number.isInteger(i));
+  }
+
+  toString() {
+    return `(${this.join(',')})`;
+  }
+
+  /**
+   * @param {Vector} ve
+   * @returns {boolean}
+   */
+  equals(ve) {
+    if (!ve || !(ve instanceof Array)) {
+      throw new Error('Not a valid Vector');
+    }
+
+    return this.length === ve.length;
+  }
```

Comment:
I believe we need to compare not only vectors length, but "each component" like mentioned in the task 

// from the task
// an equals method, to check that two vectors that have the **same components are equal**

Reply:
I noticed that you're using this method in other places to check whether two vectors have the same length, so please, don't forget to update such places.